app/services/flow/flow_state_machine.rb

```diff
@@ -85,9 +86,26 @@ def move_to_next_step
     def render_step(step, flow_session)
       @params = params
       @request = request
+      call_optional_show_step(step) and return
```

Comment:
I think `return if call_optional_show_step(step)` might be slightly clearer? the `and return` has super weird precedence and always throws me

Reply:
ah yes...thanks!